### PR TITLE
WA-199: Redesign transaction list

### PR DESCRIPTION
**Pull request by rmeissner**
Closes WA-199.

Changes proposed in this pull request:
- Apply new design to transaction list
- Add headers to transaction list

@gnosis/mobile-devs

### CODE REVIEW

File: app/src/main/java/pm/gnosis/heimdall/data/repositories/impls/DefaultGnosisSafeRepository.kt [hunk 1]
```diff
@@ -214,7 +215,13 @@ class DefaultGnosisSafeRepository @Inject constructor(
...
             }
 
-    override fun observeTransactionDescriptions(address: Solidity.Address): Flowable<List<String>> = descriptionsDao.observeDescriptions(address)
+    override fun observePendingTransactions(address: Solidity.Address): Flowable<List<TransactionStatus>> =
+        descriptionsDao.observePendingTransaction(address)
+            .map { it.map { TransactionStatus(it.id, it.timestamp, true) } }
+
+    override fun observeSubmittedTransactions(address: Solidity.Address): Flowable<List<TransactionStatus>> =
+        descriptionsDao.observeSubmittedTransaction(address)
+            .map { it.map { TransactionStatus(it.id, it.timestamp, false) } }
```
Comment (fmrsabino) [inline app/src/main/java/pm/gnosis/heimdall/data/repositories/impls/DefaultGnosisSafeRepository.kt]: I'd specify the scheduler here (for both methods)

File: app/src/main/java/pm/gnosis/heimdall/ui/safe/details/transactions/SafeTransactionsViewModel.kt [hunk 2]
```diff
@@ -31,44 +29,63 @@ class SafeTransactionsViewModel @Inject constructor(
...
+                        combined += AdapterEntry.Transaction(it.id)
+                    }
+                    combined
+                }
+            )
+                .scanToAdapterData(initialData = cachedData)
+                .doOnNext { cachedData = it }
                 .mapToResult()
-        } ?: Flowable.empty<Result<Adapter.Data<String>>>()
+        } ?: Flowable.empty<Result<Adapter.Data<AdapterEntry>>>()
```
Comment (fmrsabino) [inline app/src/main/java/pm/gnosis/heimdall/ui/safe/details/transactions/SafeTransactionsViewModel.kt]: Scheduler also missing here (when added to the `safeRepository.observePendingTransactions` or to `safeRepository.observeSubmittedTransactions` it should run on that scheduler though...)

File: app/src/main/java/pm/gnosis/heimdall/ui/safe/details/transactions/SafeTransactionsViewModel.kt [hunk 3]
```diff
@@ -31,44 +29,63 @@ class SafeTransactionsViewModel @Inject constructor(
...
+    override fun loadTransactionInfo(id: String): Single<TransactionInfo> =
+        transactionInfoRepository.loadTransactionInfo(id)
+
+    override fun loadTokenInfo(token: Solidity.Address): Single<ERC20Token> =
+        if (token == ERC20Token.ETHER_TOKEN.address)
+            Single.just(ERC20Token.ETHER_TOKEN)
+        else
+            tokenRepository.loadToken(token)
+                .onErrorResumeNext {
+                    tokenRepository.loadTokenInfo(token).firstOrError()
```
Comment (fmrsabino) [inline app/src/main/java/pm/gnosis/heimdall/ui/safe/details/transactions/SafeTransactionsViewModel.kt]: It's a bit confusing that we have `loadToken` for something that is loaded locally and `loadTokenInfo` for a remote call 🤔

Comment (rmeissner) [reply app/src/main/java/pm/gnosis/heimdall/ui/safe/details/transactions/SafeTransactionsViewModel.kt]: hmm, we can rename that later :D has always bin like that. Maybe we should also refactor this into one method with a boolean that indicates if it should try fetching from the network. will change that with WA-420

File: app/src/main/java/pm/gnosis/heimdall/ui/safe/details/transactions/SafeTransactionsViewModel.kt (hunk 2 above)
Comment (rmeissner) [reply app/src/main/java/pm/gnosis/heimdall/ui/safe/details/transactions/SafeTransactionsViewModel.kt]: added in the Repository

**Pull request merged**